OpenRA.Game/Map/Map.cs

```diff
@@ -706,13 +714,28 @@ public int FacingBetween(CPos cell, CPos towards, int fallbackfacing)
 			AllCells = new CellRegion(TileShape, tl, br);
 		}
 
-		public void ResizeCordon(int left, int top, int right, int bottom)
+		public void SetBounds(MPos tl, MPos br)
 		{
-			Bounds = Rectangle.FromLTRB(left, top, right, bottom);
+			// The tl and br coordinates are inclusive, but the Rectangle
+			// is exclusive.  Pad the right and bottom edges to match.
+			Bounds = Rectangle.FromLTRB(tl.U, tl.V, br.U + 1, br.V + 1);
+			CellsInsideBounds = new CellRegion(TileShape, tl.ToCPos(this), br.ToCPos(this));
+
+			// Directly calculate the projected map corners in world units avoiding unnecessary
+			// conversions.  This abuses the definition that the width of the cell is always
+			// 1024 units, and that the height of two rows is 2048 for classic cells and 1024
+			// for diamond cells.
```

Comment:
[SA1512] A single-line comment must not be followed by a blank line. To ignore this error when commenting out a line of code, begin the comment with '////' rather than '//'.